File: mobile_h5/src/types/conversation.ts
import type {
  MessageType,
  TaskStatusContent,
  ErrorContent,
  SummaryCardContent,
  TickerResolutionContent,
} from '@/types/messageTypes'

export type { MessageType }

export type ConversationStatus =
  | 'idle'
  | 'collecting_inputs'
  | 'ready_to_analyze'
  | 'analyzing'
  | 'report_ready'
  | 'report_explaining'
  | 'failed'

/** @deprecated 请使用 MessageType（来自 @/types/messageTypes） */
export type ConversationMessageType = MessageType

export interface TaskProgress {
  status?: string
  stageId?: string
  nodeId?: string
  displayState?: 'pending' | 'active' | 'stalled' | 'done' | 'failed' | string
  currentStep?: string
  message?: string
  elapsedTime?: number
  remainingTime?: number
}

export interface ConversationSummary {
  id: string
  title: string
  status: ConversationStatus
  updatedAt: string
}

export interface ConversationMessage {
  id: string
  role: 'user' | 'assistant' | 'system'
  messageType: MessageType
  content:
    | string
    | TaskStatusContent
    | ErrorContent
    | SummaryCardContent
    | TickerResolutionContent
  createdAt: string
}

export interface ConversationDetail extends ConversationSummary {
  messages: ConversationMessage[]
  taskProgress?: TaskProgress | null
}

export interface CreateConversationRequest {
  title?: string
}

export interface PostConversationMessageRequest {
  message: string
}

export interface PostConversationMessageResponse {
  messages: ConversationMessage[]
}

File: ta_service/models/conversation.py
from __future__ import annotations

from pydantic import BaseModel, Field

from ta_service.models.message_types import MessageType


class TaskProgress(BaseModel):
    """分析任务进度信息，仅在会话状态为 analyzing 时由后端填充。"""

    currentStep: str | None = None
    message: str | None = None
    elapsedTime: int | None = None
    remainingTime: int | None = None


class ConversationSummary(BaseModel):
    id: str
    title: str
    status: str
    updatedAt: str


class ConversationMessage(BaseModel):
    id: str
    role: str
    messageType: MessageType = Field(default=MessageType.TEXT)
    content: dict | str
    createdAt: str


class ConversationDetail(ConversationSummary):
    messages: list[ConversationMessage] = Field(default_factory=list)
    taskProgress: TaskProgress | None = None


class CreateConversationRequest(BaseModel):
    title: str | None = None


class PostConversationMessageRequest(BaseModel):
    message: str


class PostConversationMessageResponse(BaseModel):
    messages: list[ConversationMessage] = Field(default_factory=list)
